### DISCUSSION TITLE
Can one file have multiple maintainers?

### DISCUSSION THREAD

**Discussion by MaxBromberg** [nicolaasburgers/definition-library-releases#38] [Q&A]
To my understanding, having **multiple people's** updates of the same file reflected in different annotated versions is the main draw of gh version control. 
However, it seems only a file's maintainer can update it's version, and other users must fork it. Is it not possible for multiple users to update the same file, without forking it?

Best, 
Max

**Comment by nicolaasburgers** [comment]
Hi Max

It's true that the number of maintainers is limited to 1 (right now at least) but that's essentially just 1 at a time rather than a full restriction. 

You can enable others to make changes to a published definition by ceding ownership of it by the use of the "Change Maintainer" button in the Import window. 

If you're the current owner (maintainer) of a definition then when you select it in the Import window you'll see that button appear and when you press it you'll see a list of names that you can select as the new maintainer. Once they're the maintainer they can publish a change and then optimally give control back to you, or anyone else who needs to make a change.

This is one way to restrict the complexity of multiple owners, which could necessitate the ability to compare and merge different versions of a definition together right before publishing.

It's like the proverbial conch (whoever's holding the conch in a meeting can speak).

I thought this was a viable workaround for now that would be good value, in that it would involve just a few extra clicks and some comms between colleagues, leaving me able to focus on developing other features. Happy to hear your thoughts on this though.

**Reply by nicolaasburgers** [reply]
Sorry for the delay in replying by the way!

**Reply by avanboxem** [reply]
Hello Nicolaas

Thanks for your explanation. 
We're working in a small team where it would be very usefull to have multiple simultaneous maintainers. 
Is this something you are looking to implement? Or is there a way around this for now? 

Thanks

**Reply by nicolaasburgers** [reply]
Hi @avanboxem

The workaround for now is to switch maintainer to whoever wants to maintain it - so if person A's the owner and person B wants to make a change, B can let A know and A can use the Change Maintainer button to switch ownership to B (the effect is immediate); if another person C wants to make the next change then they ask B, and so on.  

This way is slightly inconvenient but it does avoid version conflicts until I implement a better solution, which, to be frank, isn't in my immediate roadmap.

I'd be open to exploring options which could allow this but which don't involve merging differences between the changes you just made with the changes a colleague just published - because comparison and merging could get a bit complex or time-consuming to implement.  Feel free to suggest some ideas for this in the Discussions board. 

Cheers
